src/main/java/org/folio/circulation/resources/RequestHoldShelfClearanceResource.java

```diff
@@ -0,0 +1,288 @@
+                                                                                List<String> itemIds) {
+    return itemIds.stream()
+      .filter(Objects::nonNull)
+      .map(itemId -> {
+        final Result<CqlQuery> servicePointQuery = exactMatch(SERVICE_POINT_ID_KEY, servicePointId);
+        final Result<CqlQuery> itemIdQuery = CqlQuery.exactMatch(ITEM_ID_KEY, itemId);
+        final Result<CqlQuery> notEmptyDateQuery = CqlQuery.exactNotEmpty(REQUEST_CLOSED_DATE_KEY);
+        final Result<CqlQuery> statusQuery = CqlQuery.exactMatchAny(STATUS_KEY,
+          Arrays.asList(CLOSED_PICKUP_EXPIRED.getValue(), CLOSED_CANCELLED.getValue()));
+
+        Result<CqlQuery> cqlQueryResult = servicePointQuery
+          .combine(itemIdQuery, CqlQuery::and)
+          .combine(statusQuery, CqlQuery::and)
+          .combine(notEmptyDateQuery, CqlQuery::and)
+          .map(q -> q.sortBy(descending(REQUEST_CLOSED_DATE_KEY)));
+
+        return findRequestsByCqlQuery(client, cqlQueryResult);
+      }).map(CompletableFuture::join)
+      .collect(Collectors.toList());
+  }
+
+  private List<Request> getFirstRequestFromList(List<Result<MultipleRecords<Request>>> multipleRecordsList) {
+    return multipleRecordsList.stream()
+      .map(r -> r.value().getRecords().stream().findFirst())
+      .filter(Optional::isPresent)
+      .map(Optional::get)
+      .collect(Collectors.toList());
+  }
+
+  private CompletableFuture<Result<MultipleRecords<Request>>> findRequestsByCqlQuery(CollectionResourceClient client,
+                                                                                     Result<CqlQuery> cqlQueryResult) {
+    return cqlQueryResult
+      .after(query -> client.getMany(query, PAGE_LIMIT))
+      .thenApply(result -> result.next(this::mapResponseToRequest));
+  }
+
+  private Result<List<Result<Request>>> fetchItemToRequest(Result<List<Request>> requests,
+                                                           ItemRepository itemRepository) {
+    return requests.map(r -> r.stream()
+      .map(request -> fetchItem(itemRepository, request))
+      .map(CompletableFuture::join)
+      .collect(Collectors.toList()));
+  }
+
+  private Result<JsonObject> mapResultToJson(Result<List<Result<Request>>> requests) {
+    return requests
+      .map(resultList ->
+        Result.combineAll(resultList).map(r -> r.stream()
+          .map(result -> new RequestRepresentation().extendedRepresentation(result))
+          .collect(Collector.of(JsonArray::new, JsonArray::add, JsonArray::add))))
+      .next(r -> r.next(jsonArray -> Result.succeeded(new JsonObject()
+        .put(REQUESTS_KEY, jsonArray)
+        .put(TOTAL_RECORDS_KEY, jsonArray.size()))));
+  }
+
+  private CompletableFuture<Result<Request>> fetchItem(ItemRepository itemRepository, Request request) {
+    return CompletableFuture.completedFuture(Result.succeeded(request))
+      .thenComposeAsync(result -> result.combineAfter(itemRepository::fetchFor, Request::withItem));
```

Comment:
Fetching an item using the item repository fetches (at least) the holdings record and instances, please update the module permissions for the endpoint to reflect this.

Reply:
Fixed